UI/CUIControls.h

```diff
@@ -594,4 +595,58 @@ class RotatingGraphic : public GG::StaticGraphic {
     float   m_phase_offset;
 };
 
+class CUIToggleButton;
+
+/** \brief Represents a CUIToggleButton
+ * 
+ * Renders a SubTexture depending on the checked state of the CUIToggleButton.
+ * When the \a button has single texture and is unchecked, it renders the same icon recolored towards grey.
+ * If a second SubTexture is assigned to the \a button, it will render the appropriate SubTexture.
+ * When a mouse pointer hovers over this button, the opposing SubTexture and color are used and highlighted
+ */
+class CUIToggleRepresenter : public GG::StateButtonRepresenter {
+public:
+    /** \name Structors */ //@{
+    CUIToggleRepresenter(boost::shared_ptr<GG::SubTexture> icon, const GG::Clr& highlight_clr); ///< ctor
+    CUIToggleRepresenter(boost::shared_ptr<GG::SubTexture> unchecked_icon, const GG::Clr& unchecked_clr,
+                         boost::shared_ptr<GG::SubTexture> checked_icon, const GG::Clr& checked_clr); ///< ctor
+    //@}
+
+    /** \name Mutators */ //@{
+    virtual void                        RenderUnchecked(const CUIToggleButton& button) const;
+    virtual void                        RenderChecked(const CUIToggleButton& button) const;
+    //@}
+
+private:
+    boost::shared_ptr<GG::SubTexture>   m_unchecked_icon;
+    boost::shared_ptr<GG::SubTexture>   m_checked_icon;
+    GG::Clr                             m_unchecked_color;
+    GG::Clr                             m_checked_color;
+};
+
+/** \brief Toggled StateButton
+ * 
+ * This StateButton can utilize a separate SubTexture and
+ * With the default single texture it renders the same icon recolored towards grey.
+ * If a second SebTexture is assigned, it will render the appropriate SubTexture.
+ * When a mouse pointer hovers over this button, the opposing SubTexture and color are used and highlighted
+ */
+class CUIToggleButton : public GG::StateButton {
+public:
+    /** \name Structors */ //@{
+    CUIToggleButton(const std::string& tooltip, boost::shared_ptr<CUIToggleRepresenter> representer); ///< ctor
+    //@}
+
+    /** \name Mutators */ //@{
+    void                                    SetTooltip(const std::string& tooltip);  ///< Set the text for the tooltip of this button.
+    virtual void                            Render();
+    //@}
+
+protected:
+    virtual void                            LClick(const GG::Pt& pt, GG::Flags<GG::ModKey> mod_keys);
+
+private:
+    boost::shared_ptr<CUIToggleRepresenter> m_representer;
```

Comment:
Already exists in GG::StateButton.  No need to store a reference to the representer at all.